bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractManagedProvider.java

```diff
@@ -122,10 +122,12 @@ public E update(E element) {
         String key = getKeyAsString(element);
         if (storage.get(key) != null) {
             PE persistableElement = storage.put(key, toPersistableElement(element));
-            E oldElement = toElement(key, persistableElement);
-            notifyListenersAboutUpdatedElement(oldElement, element);
-            logger.debug("Updated element {} in {}.", key, this.getClass().getSimpleName());
-            return oldElement;
+            if (persistableElement != null) { // redundant but makes null reference check happy.
```

Comment:
This is not redundant.
The storage member could be changed between `storage.get(key)` and `storage.put(key, ...`.

Or am I wrong? I don't see any synchronized statements that ensure storage isn't changed.
So, I agree with the change but IMHO the comment should be removed.

Reply:
Yes, you are right. The comment actually refers only to `storage.get(key)` but parallel modification does indeed require this call. I will remove the comment.